src/types.ts

```diff
@@ -37,3 +42,60 @@ export type InfuraJsonRpcSupportedNetwork =
   | 'linea-goerli'
   | 'linea-sepolia'
   | 'linea-mainnet';
+
+/**
+ * The interface for a service class responsible for making a request to an RPC
+ * endpoint.
+ */
+export type AbstractRpcService = {
```

Comment:
We could get this from `@metamask/network-controller`, but that would create a circular dependency.